backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/types/impl/ElasticsearchIndexValueFieldType.java

```diff
@@ -29,13 +34,19 @@ public class ElasticsearchIndexValueFieldType<F>
 	private final PropertyMapping mapping;
 
 	private final Consumer<PropertyMappingIndexSettingsContributor> indexSettingsContributor;
+	private final ProjectionConverter<?, ?> rawProjectionConverter;
+	private final DslConverter<?, ?> rawDslConverter;
+
 
 	public ElasticsearchIndexValueFieldType(Builder<F> builder) {
 		super( builder );
 		this.elasticsearchTypeAsJson = builder.elasticsearchTypeAsJson();
 		this.codec = builder.codec;
 		this.mapping = builder.mapping;
 		this.indexSettingsContributor = builder.indexSettingsContributor;
+
+		this.rawProjectionConverter = ProjectionConverter.passThrough( String.class );
+		this.rawDslConverter = new DslConverter<>( String.class, RawNoParsingDslConverter.INSTANCE );
```

Comment:
This seems... wrong? Why no parsing? It means all raw values people will pass to the DSL will be rendered as strings. Never numbers, never objects... Just strings.

IMO if you want to represent raw values as strings, you should use the "userFacingGson":

* where a `DslConverter` is appropriate, you'd use `userFacingGson.fromJson(userProvidedString, JsonElement.class)` to convert a user-provided JSON-formatted String (which could reprensent a number, an object, ...) to a `JsonElement` you can integrate into your request to Elasticsearch.
* where a `ProjectionConverter` is appropriate, you'd use `userFacingGson.toJson(jsonElementFromResponse)` to convert the `JsonElement` from the response to JSON-formatted String (which could reprensent a number, an object, ...) the user can then manipulate.

Now do you want to represent raw values as strings... that's another question. I guess in the case of avg aggregations on date/numeric fields, you don't, since you want a `double` there. Maybe we need something more subtle, like:

* if we expect JsonPrimitive on the JSON side, the "raw" model will be String/Number/Boolean/Character, depending on the case
* if we expect JsonArray on the JSON side, `List<T>` with T being determined using the same rules
* if we expect JsonObject on the JSON side... I don't know. `JsonObject` maybe? And document this might change in the future.

Reply:
BTW if you want to just return `JsonElement`, document this as unstable and change the behavior when we integrate the "raw avg as double", that's fine by me.

Reply:
I mean ... you had a reason for not wanting to expose the Gson 😄 so unless we reaaaaaaaly need to let's work around it 😃

Reply:
I've pushed a change so that we work with strings but through a Gson... now if we want to do this 

> if we expect JsonPrimitive on the JSON side, the "raw" model will be String/Number/Boolean/Character, depending on the case

hmm IDK ... so we'd update the codec to be `interface ElasticsearchFieldCodec<F, *V*> ` and those couple methods I've added would become:
```
	JsonElement fromJsonStringToElement(V value);

	V fromJsonElementToString(JsonElement value);
```
(obviously with a different name 😄). If we consider the `.avg( "myDateTimeField", Double.class, ModelValue.RAW)` it may work if we just have something like:

```java
double d = DoubleCodec.fromJsonElementToString(value); // just reusing the same name as above...
if (valuemodel == RAW) {
    return d;
} else {
  // I think Fabio had something in his PR to convert double to the field type....
  return actualFieldCodec.smth(d);
}
```

maybe it'll work 😃